lib/base-compiler.ts

```diff
@@ -1749,6 +1756,18 @@ export class BaseCompiler implements ICompiler {
         return this.runCompiler(compiler, options, inputFilename, execOptions);
     }
 
+    protected maskArgumentsForCompilation(args: string[]): string[] {
+        if (this.executionType === 'nsjail' || this.executionType === 'cewrapper') {
```

Comment:
I'd really love to not leak this outside of the exec code itself. Is there a way this functionality could move to the `exec` module, so at least all this horrid path manipulation is in one place (and tested) instead of sprinkled in a few places.

Is there a reason this isn't already handled by the rewriting in the exec code? It seems it's added to `compilationOptions` -- is that not already transformed when we go to `exec` it or is that someting else?

Reply:
Definitely not no, this is the live situation:
![Screenshot from 2024-02-06 12-26-10](https://github.com/compiler-explorer/compiler-explorer/assets/442630/f1e12ebb-c565-41b3-af20-4d0c5d54b20d)

And this is without any libraries.

With more libraries it gets incredibly unreadable and basically unusable noise.

But I'm happy to extract the masking code to a new PR if you want some time to think about a better solution

Reply:
Actually I can't, it's already too engrained into things like https://github.com/compiler-explorer/compiler-explorer/blob/main/lib/parsers/asm-parser.ts#L387
And now it's just "wrong" (though it still works there, because the filenames during compilation are called either `/app/etc` or `/tmp/compiler-explorer-etc-etc/etc` there) - but we don't have dirPath there, so it will always need to be some regular expression you want to rewrite that as well.

So for now I'm just cleaning up and merging

Reply:
thanks for checking, i concur :) merge once you're happy!